test/Interop/Cxx/operators/Inputs/member-inline.h

```diff
@@ -39,4 +39,67 @@ struct HasDeletedOperator {
   void operator!=(HasDeletedOperator) const = delete;
 };
 
+struct ReadWriteIntArray {
+private:
+  int values[5] = { 1, 2, 3, 4, 5 };
+
+public:
+  const int &operator[](int x) const {
+    return values[x];
+  }
+  int &operator[](int x) {
+    return values[x];
+  }
+};
+
+struct ReadOnlyIntArray {
+private:
+  int values[5] = { 1, 2, 3, 4, 5 };
+
+public:
+  ReadOnlyIntArray() {}
+  ReadOnlyIntArray(const ReadOnlyIntArray &other) {}
+
+  const int &operator[](int x) const {
+    return values[x];
+  }
+};
+
+struct WriteOnlyIntArray {
+private:
+  int values[5] = { 1, 2, 3, 4, 5 };
+
+public:
+  int &operator[](int x) {
+    return values[x];
+  }
+};
+
+struct DifferentTypesArray {
+private:
+  int values[3] = { 1, 2, 3 };
+  double doubleValues[2] = { 1.5, 2.5 };
+
+public:
+  const int &operator[](int x) const {
+    return values[x];
+  }
+  int &operator[](int x) {
+    return values[x];
+  }
+  const double &operator[](double x) const {
+    return doubleValues[x == 0.0];
+  }
+};
+
+struct NonReferenceReadIntArray {
+private:
+  int values[3] = { 1, 2, 3 };
+
+public:
+  int operator[](int x) const {
```

Comment:
> I'd like to see a test that we import templated subscript operators correctly.

We can't – Swift doesn't support generic subscripts :( 
But I've added a new test for a subscript in a generic struct.

> Also, can you add a test for address only types?

Yeap, I indended to make `ReadOnlyIntArray` address-only, but something went wrong...
Fixed.

Reply:
> We can't – Swift doesn't support generic subscripts :(
> But I've added a new test for a subscript in a generic struct.

Thanks! However, as of Swift 4, [generic subscripts are supported](https://swift.godbolt.org/z/1jfbbo). Even if they weren't supported, it would be good to add a test for templated subscript operators so that we make sure we don't crash (now or in the future) when importing them.

Reply:
Oh my bad, didn't know that. Templated subscripts aren't currently imported, but I've added a module interface test to make sure we don't crash.